// flow-launcher-helper/lib/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Flow = void 0;
class Flow {
    constructor(defaultIconPath) {
        this.methods = {};
        this.data = JSON.parse(process.argv[2]);
        this.defaultIconPath = defaultIconPath;
        this.showResult = this.showResult.bind(this);
        this.run = this.run.bind(this);
        this.on = this.on.bind(this);
    }
    get method() {
        return this.data.method;
    }
    get params() {
        return this.data.parameters[0];
    }
    get settings() {
        return this.data.settings;
    }
    on(method, callbackFn) {
        this.methods[method] = callbackFn;
    }
    showResult(...result) {
        const generateResult = () => {
            const results = [];
            for (const r of result) {
                results.push({
                    Title: r.title,
                    Subtitle: r.subtitle,
                    JsonRPCAction: {
                        method: r.method,
                        parameters: r.params,
                    },
                    IcoPath: r.iconPath || this.defaultIconPath,
                });
            }
            return { result: results };
        };
        return console.log(JSON.stringify(generateResult()));
    }
    run() {
        this.data.method in this.methods && this.methods[this.data.method]();
    }
}
exports.Flow = Flow;
